.github/workflows/simple.yaml

```diff
@@ -0,0 +1,138 @@
+---
+name: Simple workflow
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+# For reference, this is how the final list of steps will look like:
+# env:
+#   ENABLED_STEPS: |
+#     checkout_repository
+#     setup_r
+#     setup_dependencies
+#     get_package_name
+#     build_r_package
+#     run_r_cmd_check
+#     upload_artifact
+
+# When asked to, please add the steps from the list above:
+env:
+  ENABLED_STEPS: |
+    checkout_...
+    setup_...
+    setup_...
+    get_package_...
+    build_r_...
+    run_r_...
+    upload_...
+
+jobs:
+  # First job.
+  hello-world-from-r:
+    name: Hello, world!
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup R 🔧
+        uses: r-lib/actions/setup-r@v2
+
+      - name: Hello world! 💬
+        run: cat(paste0("Hello ", Sys.getenv("NAME"), "! 👋"))
+        shell: Rscript {0}
+        env:
+          NAME: "Your name"
+
+  # Second job - by default jobs will run in parallel, unless some dependencies
+  # are introduced between them.
+  build-and-check:
+    name: Build R package
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo 🛎
+        # We're using an official GitHub Action.
+        uses: actions/checkout@v4.1.1
+        if: contains(env.ENABLED_STEPS, 'checkout_repository')
+        with:
+          # Action inputs (arguments/parameters).
+          # In this case we clone the repository to the directory called the same
+          # as the repository name.
+          path: ${{ github.event.repository.name }}
+
+      - name: Setup R 🔧
+        if: contains(env.ENABLED_STEPS, 'setup_r')
+        # We're using a third-party GitHub Action.
+        uses: r-lib/actions/setup-r@v2
+
+      - name: Setup R dependencies 🔧
+        # We're using a third-party GitHub Action.
+        uses: r-lib/actions/setup-r-dependencies@v2
+        if: contains(env.ENABLED_STEPS, 'setup_dependencies')
+        with:
+          extra-packages: any::rcmdcheck
+          working-directory: ${{ github.event.repository.name }}
+
+      # Get package name and version from DESCRIPTION so that we'll know
+      # what's the name of the tar.gz file with the built package.
+      # PKGBUILD environment variable is saved as an environment variable so this
+      # information can be accessed by subsequent steps.
```

Comment:
what would be better: save to envvar or return value from step?
we are already using envvar in the condition (i.e. `ENABLED_STEPS`) so maybe it would be good for demonstrative purposes to show both in action? WDYT?

Reply:
Yeah, no problem, let me change that.